pyflakes/checker.py

```diff
@@ -453,6 +453,17 @@ def getParent(self, node):
             if not hasattr(node, 'elts') and not hasattr(node, 'ctx'):
                 return node
 
+    def has_parent_type(self, node, cls):
+        """Check if node has any parent of type cls."""
+        while node:
+            try:
+                node = self.getParent(node)
```

Comment:
`getParent` also skips some parents when `if not hasattr(node, 'elts') and not hasattr(node, 'ctx')` is not True.

However, not isnt really needed in this case, and inline-ing `getParent` will like be faster, or at least more clear anyway.

Also, this routine is probably better named `has_ancestor_type`.